src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -410,10 +414,50 @@ export default class SurvivalChart extends React.Component<ISurvivalChartProps,
         return this.victoryLegendData.length > 0;
     }
 
+    @observable.shallow zoomDomain: any = {x: [0, this.initialRangeMaxValue]};
+
+    @autobind
+    private onZoom(domain: any, props: any) {
+        this.zoomDomain = {x: domain.x};
+    }
+
+    @autobind
+    private updateDomain(domain: any) {
+        this.zoomDomain = {x: domain.x};
+    }
+
+    @computed get maximumDataMonthValue() {
+        let max = Number.MIN_VALUE;
+        _.forEach(this.sortedGroupedSurvivals, (survivals) => {
+            max = Math.max(max, survivals[survivals.length - 1].months)
+        });
+        return Math.ceil(max);
+    }
+
+    @computed get initialRangeMaxValue() {
+        if (this.maximumDataMonthValue < SURVIVAL_CHART_INITIAL_SHOWN_MONTHS) {
+            return this.maximumDataMonthValue;
+        }
+        return SURVIVAL_CHART_INITIAL_SHOWN_MONTHS;
+    }
+
+    @computed get additionalProps() {
+        if (this.props.showRangeSelector) {
```

Comment:
getter should always return something.  undefined would work here.  i'm surprised that ts doesn't object to this